crates/matrix-sdk/src/client/mod.rs

```diff
@@ -210,6 +210,10 @@ pub(crate) struct ClientLocks {
     /// outside the `OlmMachine`.
     #[cfg(feature = "e2e-encryption")]
     pub(crate) crypto_store_generation: Arc<Mutex<Option<u64>>>,
+
+    /// Ensure that only one service is syncing at a time. Otherwise, they will
+    /// override each other's updates because they batch changes.
+    pub(crate) sync_service_lock: Mutex<()>,
```

Comment:
So we have a `ClientLocks` data structure where to try to put all the locks. Independently of that, the `BaseClient` does already have a `sync_lock`, can we reuse it instead?